### ISSUE TITLE
ANGLE doesn't build on Windows

### ISSUE THREAD

**Issue by robmikh** [Microsoft/vcpkg#3690]
When attempting to build ANGLE on Windows, there are errors related to not having control flow guard enabled in MSVC:

```
angle_preprocessor.lib(ExpressionParser.cpp.obj) : error LNK2001: unresolved external symbol _guard_dispatch_icall
angle_preprocessor.lib(Input.cpp.obj) : error LNK2001: unresolved external symbol _guard_dispatch_icall
angle_common.lib(MemoryBuffer.cpp.obj) : error LNK2001: unresolved external symbol _guard_dispatch_icall
angle_common.lib(string_utils.cpp.obj) : error LNK2001: unresolved external symbol _guard_dispatch_icall
angle_preprocessor.lib(Macro.cpp.obj) : error LNK2001: unresolved external symbol _guard_dispatch_icall
angle_preprocessor.lib(DirectiveParser.cpp.obj) : error LNK2001: unresolved external symbol _guard_dispatch_icall
angle_preprocessor.lib(MacroExpander.cpp.obj) : error LNK2001: unresolved external symbol _guard_dispatch_icall
angle_preprocessor.lib(Token.cpp.obj) : error LNK2001: unresolved external symbol _guard_dispatch_icall
angle_translator.lib(UtilsHLSL.cpp.obj) : error LNK2001: unresolved external symbol _guard_dispatch_icall
angle_translator.lib(UniformHLSL.cpp.obj) : error LNK2001: unresolved external symbol _guard_dispatch_icall
angle_preprocessor.lib(DiagnosticsBase.cpp.obj) : error LNK2001: unresolved external symbol _guard_dispatch_icall
angle_preprocessor.lib(Preprocessor.cpp.obj) : error LNK2001: unresolved external symbol _guard_dispatch_icall
angle_translator.lib(RemoveSwitchFallThrough.cpp.obj) : error LNK2001: unresolved external symbol _guard_dispatch_icall
angle_translator.lib(SearchSymbol.cpp.obj) : error LNK2001: unresolved external symbol _guard_dispatch_icall
angle_translator.lib(StructureHLSL.cpp.obj) : error LNK2001: unresolved external symbol _guard_dispatch_icall
angle_translator.lib(TextureFunctionHLSL.cpp.obj) : error LNK2001: unresolved external symbol _guard_dispatch_icall
angle_translator.lib(VersionGLSL.cpp.obj) : error LNK2001: unresolved external symbol _guard_dispatch_icall
angle_translator.lib(ASTMetadataHLSL.cpp.obj) : error LNK2001: unresolved external symbol _guard_dispatch_icall
angle_translator.lib(BuiltInFunctionEmulatorHLSL.cpp.obj) : error LNK2001: unresolved external symbol _guard_dispatch_icall
angle_translator.lib(FlagStd140Structs.cpp.obj) : error LNK2001: unresolved external symbol _guard_dispatch_icall
angle_translator.lib(OutputGLSLBase.cpp.obj) : error LNK2001: unresolved external symbol _guard_dispatch_icall
angle_translator.lib(OutputESSL.cpp.obj) : error LNK2001: unresolved external symbol _guard_dispatch_icall
angle_translator.lib(OutputGLSL.cpp.obj) : error LNK2001: unresolved external symbol _guard_dispatch_icall
(there's a bunch more of these...)
```

I have a fix in this branch: [robmikh/vcpkg/anglefix](https://github.com/robmikh/vcpkg/tree/anglefix)

The fix amounts to moving the /guard:cf flag from the linker flags to the compile options. Note that this exacerbates #3321 and exacerbates the issues that ANGLE currently doesn't build on Linux either (tested on Ubuntu 18.04).

**Comment by robmikh** [comment]
I tried to get it building on Ubuntu 18.04, but I ran into a couple of issues.

First, there's a dependency on `X11/Xlib.h`. To get past that, I had to run `sudo apt-get install libx11-dev`. I'm not that familiar with how vcpkg handles dependencies, so I'm not sure how to fix that one. 

Second, it doesn't seem that the cmake file had anything other than Windows in mind. By default, it pulls in everything related to D3D and sets ANGLE_ENABLE_D3D11. It also links against the d3d version of the renderer, and the opengl renderer is limited to WIN32. 

Finally, the way it pulls in the common angle files will pull in Windows specific files as well. Strangely it also pulls in Linux and MacOS files... but it still builds in Windows (after I applied the fix from above)?

I might investigate this further at a later time, but it would be great if the Windows build could be unblocked for now. :)
